packages/core/src/components/dialog/multistepDialog.tsx

```diff
@@ -103,7 +116,7 @@ export class MultistepDialog extends AbstractPureComponent2<MultistepDialogProps
             !prevProps.isOpen &&
```

Comment:
with the changes I suggested above, we can get rid of the `prevProps.resetOnClose === undefined` check. I think that will make this easier to read (it took me a second to figure out what's going on)...

in addition, I think we should check if `props.initialStepIndex` changed, and reset the state in that case too...

```ts
if (
    (prevProps.resetOnClose && !prevProps.isOpen && this.props.isOpen) ||
    (prevProps.initialStepIndex !== this.props.initialStepIndex)
) {
```

Reply:
hm... it feels weird to change the current index if the `initialStepIndex` changes while it's still open?

Reply:
what about
```
if (
            (prevProps.resetOnClose || prevProps.initialStepIndex !== this.props.initialStepIndex) &&
            !prevProps.isOpen &&
            this.props.isOpen
        ) {
```